### DISCUSSION TITLE
Will quickwit support storing Prometheus metrics?

### DISCUSSION THREAD

**Discussion by batmilkyway** [quickwit-oss/quickwit#3843] [New features]
As an SRE/platform/cloud/whatever engineer I'd like to self-host my observability tools to reduce costs (DataDog, New Relic, etc are quite costly). One of the pain points with self-hosting is that tools are either fragmented, or require heavy infrastructure investment. 

For example, the LGTM stack requires deploying Loki, Tempo, and Mimir (or other Prometheus compatible store) to capture logs, metrics, and traces.

Other projects like SigNoz or Uptrace, which do support a unified observability store, do it with ClickHouse, which can be pretty resource hungry and also incurs operational overhead.

Quickwit already supports logs and traces, will it also support metrics?

**Comment by guilload** [comment]
Hello @batmilkyway,

We want to support metrics eventually, but this will require much work because inverted indexes are not best suited for storing/querying metrics. However, I doubt we can work on this anytime soon because we are already quite busy with other higher-priority features.

**Reply by devopstoday11** [reply]
@guilload 
If community wants to contribute the Metrics support, how can we? Are there detailed step by step guidelines on how that particular feature can be developed? Maybe using that guidelines, some of us can contribute.

**Comment by batmilkyway** [comment]
Thank you for the reply, looking forward to when it lands!

**Comment by andrewbanchich** [comment]
Is there a tracking issue for this I can subscribe to @fulmicoton ?

**Reply by rdettai** [reply]
Not really, and right now this is still not prioritized. I thinks subscribing to this discussion should be enough to notify you if things start moving forward.
